### ISSUE TITLE
`DoTransition` method with payload

### ISSUE THREAD

**Issue by dk13danger** [andygrunwald/go-jira#57]
Hi!
As i can see in code you don't realize facility for injecting payload with performing transition yet?
Here: https://github.com/andygrunwald/go-jira/blob/master/issue.go#L648
Do you have plan for updating? When this will be implemented?

**Comment by andygrunwald** [comment]
Hey @dk13danger,

indeed. You are right.
As stated in the [docs](https://docs.atlassian.com/jira/REST/cloud/#api/2/issue-doTransition) there are a lot more possibilities for the DoTransistion method.
Right now we are only supporting

```go
payload := CreateTransitionPayload{
	Transition: TransitionPayload{
		ID: transitionID,
	},
}
```

> Do you have plan for updating? When this will be implemented?
At the moment i don't have specific plans for updating this. If you or someone else would be happy to apply a [Pull Request](https://help.github.com/articles/about-pull-requests/) for it, i would be happy to merge this.
Would you mind to implement this?

An alternative way is that you implement the functionality yourself via [Call a not implemented API endpoint](https://github.com/andygrunwald/go-jira#call-a-not-implemented-api-endpoint).

**Comment by dk13danger** [comment]
Thx, @andygrunwald )
I know, of course, about `not implemented API endpoint` workaround)
I switched for my project on `php`, that's why i don't lack in this feature right now.
But i very want to do PR in future, when i'll return to golang :)

**Comment by andygrunwald** [comment]
Thats nice to hear @dk13danger! Good luck with your project :)
I will leave this issue open until this is fixed.

**Comment by andygrunwald** [comment]
Implemented by @yanolab in #72
